modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C

```diff
@@ -0,0 +1,513 @@
+    if (!_ring_block_ids.empty() &&
+        _ring_block_ids.size() !=
+            (_ring_intervals.size() + (unsigned int)(_ring_intervals.front() != 1)))
+      paramError("ring_block_ids",
+                 "This parameter must have the appropriate size if it is provided.");
+    if (!_ring_block_names.empty() &&
+        _ring_block_names.size() !=
+            (_ring_intervals.size() + (unsigned int)(_ring_intervals.front() != 1)))
+      paramError("ring_block_names", "This parameter must have the appropriate size if it is set.");
+    for (unsigned int i = 1; i < _ring_intervals.size(); i++)
+      if (_ring_radii[i] <= _ring_radii[i - 1])
+        paramError("ring_radii", "This parameter must be strictly ascending.");
+    if (_ring_radii.back() >= _pitch / 2.0)
+      paramError("ring_radii", "Elements of this parameter must be smaller than polygon apothem.");
+  }
+  if (_has_ducts)
+  {
+    if (_duct_sizes.size() != _duct_intervals.size())
+      paramError("duct_sizes", "This parameter and duct_intervals must have the same length.");
+    if (!_duct_block_ids.empty() && _duct_block_ids.size() != _duct_intervals.size())
+      paramError("duct_block_ids",
+                 "This parameter must have the same length as duct_intervals if set.");
+    if (!_duct_block_names.empty() && _duct_block_names.size() != _duct_intervals.size())
+      paramError("duct_block_names",
+                 "This parameter must have the same length as duct_intervals if set.");
+    if (_duct_sizes_style == DuctStyle::apothem)
+      for (unsigned int i = 0; i < _duct_sizes.size(); i++)
+        _duct_sizes[i] /= std::cos(M_PI / Real(_num_sides));
+    for (unsigned int i = 1; i < _duct_sizes.size(); i++)
+      if (_duct_sizes[i] <= _duct_sizes[i - 1])
+        paramError("duct_sizes", "This parameter must be strictly ascending.");
+    if (_duct_sizes.front() <=
+        (_has_rings ? _ring_radii.back() / std::cos(M_PI / Real(_num_sides)) : 0.0))
+      paramError("duct_sizes",
+                 "This parameter must be positive and ensures no overlapping with rings.");
+    if (_duct_sizes.back() >= _pitch / 2.0 / std::cos(M_PI / Real(_num_sides)))
+      paramError("duct_sizes",
+                 "This parameter must ensure that ducts are smaller than the polygon size.");
+    if (*std::min_element(_duct_intervals.begin(), _duct_intervals.end()) <= 0)
+      paramError("duct_intervals", "Elements of this parameter must be positive.");
+  }
+}
+
+std::unique_ptr<MeshBase>
+PolygonConcentricCircleMeshGeneratorBase::generate()
+{
+  unsigned int mesh_input_counter = 0;
+  // azimuthal array used for radius radius_correction
+  std::vector<Real> azimuthal_list;
+  for (unsigned int mesh_index = 0; mesh_index < _num_sides; mesh_index++)
+  {
+    if (std::find(_sides_to_adapt.begin(), _sides_to_adapt.end(), mesh_index) !=
+        _sides_to_adapt.end())
+    {
+      // The following lines only work for hexagon; and only a hexagon needs such functionality.
```

Comment:
add a check for this in the constructor

Reply:
Only `HexagonConcentricCircleAdaptiveBoundaryMeshGenerator` assigns a non-empty `_side_to_adapt` vector. So the check is not needed.

Reply:
well someone could be wanting to add a subclass since this particular meshgenerator is very general

Reply:
We should move this parameter to this class too.